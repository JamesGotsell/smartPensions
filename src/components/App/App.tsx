import React, { useEffect, useState } from "react";
import useAllPeopleQuery from "../../data/useAllPeopleQuery";

import useAllStarShipQuery from "../../data/useAllStarShipQuery";
import { getRamdomValue } from '../../lib/utils'

import styled from "styled-components"
import {Nav} from '../Nav/Nav'
import {Home} from '../Pages/Home'
import {ResultsPage} from '../Pages/ResultsPage'
import {
    BrowserRouter as Router,
    Switch,
    Route,
  } from "react-router-dom";

interface gameData {
    player1Selection: string;
    computerSelection: string;
    characterGame: boolean;
}
export const App: React.FunctionComponent = () => {
    const Wrapper = styled.section`
            padding: 4em;
            background: papayawhip;
            height:100%;
    `;

    const Title = styled.h1`
        font-size: 1.5em;
        text-align: center;
        color: palevioletred;
        font-family: "museo", Helvetica Neue, Helvetica, sans-serif;
    `;



return (
    <>  
        <Wrapper>
              
                <Title>STARWARS TOP TRUMPS</Title>
            <Router>
                    <Nav/>
                        <Switch>
                            <Route path="/results">
                                <ResultsPage/>
                            </Route>
                        
                            <Route path="/">
                                <Home />
                            </Route>
                        </Switch>
            </Router>
        </Wrapper>
    </>
)


};
