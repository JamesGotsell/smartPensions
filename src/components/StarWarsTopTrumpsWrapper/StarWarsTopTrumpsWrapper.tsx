import React, {useEffect, useState} from "react";
import useStarShipQuery from '../../data/useStarShipQuery'
import usePersonQuery from '../../data/usePersonQuery'
import {CardWrapper} from '../CardWrapper/CardWrapper'

interface WrapperProps {
    characterGame: boolean;
    gameTitle: string;
    computerSelection: string;
    player1Selection: string;
}

interface Selection {
  firstSelection?: {},
  secondSelection?: {}
}

// need to extract data in to first selection secondSeleciton
export const StarWarsTopTrumpsWrapper: React.FC<WrapperProps> = ({characterGame, computerSelection, player1Selection, gameTitle}) => {
     // can move this in to the top trumps component
     const [selection, setSeleciton] = useState<Selection>({
      firstSelection: {},
      secondSelection: {}
     })
     const [person, PersonError, isLoadingPerson ] = usePersonQuery(player1Selection, computerSelection)
     const [starShip, errorStarShip, isLoadingStarShip] = useStarShipQuery(player1Selection,computerSelection)
 


  useEffect(() => {
      if(characterGame && person) {
          setSeleciton({
            firstSelection: person.firstSelection,
            secondSelection: person.secondSelection
          })
      }
      if(!characterGame && starShip) {
        setSeleciton({
          firstSelection: starShip.firstSelection,
          secondSelection: starShip.secondSelection
        })
      }

      if(characterGame && !person) {
        setSeleciton({
          firstSelection: {

          },
          secondSelection: {

          }
        })
      }
  
  }, [person, starShip])

 

  if (isLoadingPerson || isLoadingStarShip) return null;


  return (
    <div>
      <h1>{gameTitle}</h1>
      <CardWrapper characterGame={characterGame} firstSelection={selection.firstSelection} secondSelection={selection.secondSelection} />
    </div>
  );
};
