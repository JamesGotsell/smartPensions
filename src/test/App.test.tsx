import React from 'react';
import {  shallow } from 'enzyme';
import {App}  from '../components/App/App';
import { StarWarsTopTrumpsWrapper } from '../components/StarWarsTopTrumpsWrapper/StarWarsTopTrumpsWrapper'
//  need to mock 
test('renders the component', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find(StarWarsTopTrumpsWrapper)).toEqual(1)
});