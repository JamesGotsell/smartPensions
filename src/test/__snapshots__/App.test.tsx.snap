// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Fragment>
  <styled.section>
    <Memo(l) />
    <styled.h1>
      STARWARS TOP TRUMPS
    </styled.h1>
    <BrowserRouter>
      <Component />
      <Switch>
        <Route
          component={[Function]}
          path="/results"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
      </Switch>
    </BrowserRouter>
  </styled.section>
</Fragment>
`;
